epoch.go

```diff
@@ -1959,7 +1964,8 @@ func (e *Epoch) buildBlock() {
 
 func (e *Epoch) createBlockBuildingTask(metadata ProtocolMetadata) func() Digest {
 	return func() Digest {
-		block, ok := e.BlockBuilder.BuildBlock(e.finishCtx, metadata)
+		e.blockBuilderCtx, e.blockBuilderCancelFunc = context.WithCancel(e.ctx)
+		block, ok := e.BlockBuilder.BuildBlock(e.blockBuilderCtx, metadata)
```

Comment:
I think we need to `defer` cancel the context anyway, even if we fail building the block.